### PR TITLE
feat(bigquery): Add example for creating a external dataset

### PR DESCRIPTION
**Pull request by misiek1984**
Intending to use this snippet on the following page:
https://cloud.google.com/bigquery/docs/spanner-external-datasets

### CODE REVIEW

**Comment by Charroschavito** [comment]
No

File: bigquery/bigquery_create_dataset_external/main.tf [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+# [START bigquery_create_dataset_external]
+resource "google_bigquery_dataset" "default" {
+  dataset_id    = "my_external_dataset"
+  friendly_name = "my_external_dataset"
```
Comment (wj-chen) [inline bigquery/bigquery_create_dataset_external/main.tf:19]: Maybe "my external dataset" to show that it doesn't need to conform to the same naming rule as `dataset_id`

File: bigquery/bigquery_create_dataset_external/main.tf [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  description   = "This is a test description."
+  location      = "US"
+  external_dataset_reference {
+    external_source = "google-cloudspanner:/projects/my_project/instances/my_instance/databases/my_database"
+    connection      = ""
```
Comment (wj-chen) [inline bigquery/bigquery_create_dataset_external/main.tf:24]: A valid `connection` value is required. Maybe we can follow this example: https://registry.terraform.io/providers/hashicorp/google/6.10.0/docs/resources/bigquery_dataset#example-usage---bigquery-dataset-external-reference-aws

File: bigquery/bigquery_create_dataset_external/main.tf (hunk 1 above)
Comment (misiek1984) [reply bigquery/bigquery_create_dataset_external/main.tf]: Done.

File: bigquery/bigquery_create_dataset_external/main.tf (hunk 2 above)
Comment (misiek1984) [reply bigquery/bigquery_create_dataset_external/main.tf]: Spanner external datasets  does not require a connection. It must be empty.

File: bigquery/bigquery_create_spanner_external_dataset/main.tf [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# [START bigquery_create_spanner_external_dataset]
+resource "google_bigquery_dataset" "default" {
+  dataset_id    = "my_external_dataset"
+  friendly_name = "My external dataset"
+  description   = "This is a test description."
+  location      = "US"
+  external_dataset_reference {
+    # The full identifier of your Spanner database
+    external_source = "google-cloudspanner:/projects/my_project/instances/my_instance/databases/my_database"
+    # Must be empty
```
Comment (wj-chen) [inline bigquery/bigquery_create_spanner_external_dataset/main.tf:25]: "Must be empty for a Spanner external dataset" to be more accurate

**Review by wj-chen — APPROVED** [review]

File: bigquery/bigquery_create_spanner_external_dataset/main.tf (hunk 3 above)
Comment (misiek1984) [reply bigquery/bigquery_create_spanner_external_dataset/main.tf]: Done.

**Pull request merged**